docs/source/references/internals/graph_breakdown.rst

```diff
@@ -0,0 +1,43 @@
+Pipeline Writing (Breaking Down the Graph)
+------------------------------------------
+
+Overview
+~~~~~~~~
+
+A diagram to show the main abstractions we use to break down a graph as defined when users define a pipelines.
+
+.. image:: ../../_static/images/graph_reader_classes.png
```

Comment:
The image file is missing (should be committed).

Reply:
woops they were being .gitignored, ty for catching this